deploy/fabfile.py

```diff
@@ -33,26 +32,58 @@
 home = "/home/site"
 log = "%s/dashboard.log" % home
 current = "%s/%s/dashboard" % (home, environment)
+now = time.strftime("%Y-%m-%d", time.localtime())
 
 # principal command to run upon update
-command = "cd %s && git pull && git submodule update && jekyll build >> %s" % (current, log)
+staging = "cd %s && git pull && bundle && git submodule update --remote && ./_data/import-public.rb && git add _data/projects.yml && git commit -m 'Update data for %s' && git push && ./go build >> %s" % (current, now, log)
+production = "cd %s && git pull && ./go build >> %s" % (current, log)
 
 ## can be run on their own
 
 def start():
-  run(
-    "cd %s && forever start -l %s -a deploy/hookshot.js -p %i -b %s -c \"%s\""
-    % (current, log, port, environment, command)
-  )
+  if type(port) != int:
+    exit(1)
+
+  with cd(current):
+    if environment == 'staging':
```

Comment:
Rather than duplicating the `if environment` logic in three places, how about:

```python
if environment == 'staging':
  deploy_cmd = "cd %s ..." % (current, now, log)
elif environment == 'production':
  deploy_cmd = "cd %s ..." % (current, log)
else:
  exit(1)
```
And then none of the `stop()`/`start()`/`restart()` cases have `if` conditions?